diff --git a/lib/dolphin/gx/GXTexture.cpp b/lib/dolphin/gx/GXTexture.cpp
index 04f97f5..73ca0f2 100644
--- a/lib/dolphin/gx/GXTexture.cpp
+++ b/lib/dolphin/gx/GXTexture.cpp
@@ -120,7 +120,7 @@ void GXInitTexObjLOD(GXTexObj* obj_, GXTexFilter minFilt, GXTexFilter magFilt, f
                      float lodBias, GXBool biasClamp, GXBool doEdgeLod, GXAnisotropy maxAniso) {
   auto* obj = reinterpret_cast<GXTexObj_*>(obj_);
   const float clampedBias = std::clamp(lodBias, -4.0f, 3.99f);
-  const auto lbias = static_cast<u8>(32.0f * clampedBias);
+  const auto lbias = static_cast<u8>(static_cast<s8>(32.0f * clampedBias));
   SET_REG_FIELD(0, obj->mode0, 8, 9, lbias);
   SET_REG_FIELD(0, obj->mode0, 1, 4, magFilt == GX_LINEAR ? 1 : 0);
   SET_REG_FIELD(0, obj->mode0, 3, 5, GX2HWFiltConv[minFilt]);
@@ -174,6 +174,6 @@ void GXInitTexObjMinLOD(GXTexObj* obj_, float minLod) {
 void GXInitTexObjLODBias(GXTexObj* obj_, float lodBias) {
   auto* obj = reinterpret_cast<GXTexObj_*>(obj_);
   const float clampedBias = std::clamp(lodBias, -4.0f, 3.99f);
-  const auto lbias = static_cast<u8>(32.0f * clampedBias);
+  const auto lbias = static_cast<u8>(static_cast<s8>(32.0f * clampedBias));
   SET_REG_FIELD(0, obj->mode0, 8, 9, lbias);
 }
